haystack/preview/components/reader.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+from haystack.preview import component, Document, Answer
+from haystack.preview.pipeline import Pipeline
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    from transformers import AutoModelForQuestionAnswering
+    from tokenizers import Tokenizer, Encoding
+    import torch
+
+
+@component
+class ExtractiveReader:
+    def __init__(
+        self, reader: Union[Path, str], device: Optional[str] = None, max_seq_length: int = 384, top_k: int = 10
+    ) -> None:
+        self.reader = reader
```

Comment:
Should the reader already be validated in __init__? We can't do this completely because it would be equivalent to loading it already, but we could still do some things.